ironfish/src/blockchain/blockchain.ts

```diff
@@ -205,21 +244,58 @@ export class Blockchain<
     return Math.max(Math.min(1, progress), 0)
   }
 
+  private loadDefaultGenesisBlock = () => {
+    return Promise.resolve(IJSON.parse(genesisBlockData) as SerializedBlock<SH, ST>)
+  }
+
   async open(): Promise<void> {
     if (this.opened) return
     this.opened = true
 
     await this.db.open()
 
-    this.genesis = await this.getHeaderAtSequence(GENESIS_BLOCK_SEQUENCE)
+    const genesis = await this.getHeaderAtSequence(GENESIS_BLOCK_SEQUENCE)
+    if (genesis) {
+      this.genesis = genesis
+      this.head = this.genesis
+      this.latest = this.genesis
+    } else if (this.autoSeed) {
+      const serialized = await this.loadGenesisBlock()
+      const genesis = this.strategy.blockSerde.deserialize(serialized)
+
+      const result = await this.addBlock(genesis)
+      Assert.isTrue(result.isAdded, `Could not seed genesis: ${result.reason || 'unknown'}`)
+
+      const genesisHeader = await this.getHeaderAtSequence(GENESIS_BLOCK_SEQUENCE)
+      Assert.isNotNull(
+        genesisHeader,
+        'Added the genesis block to the chain, but could not fetch the header',
+      )
+
+      this.genesis = genesisHeader
+      this.head = this.genesis
+      this.latest = this.genesis
+    }
 
     const headHash = await this.meta.get('head')
-    if (headHash) this.head = await this.getHeader(headHash)
+    if (headHash) {
+      const head = await this.getHeader(headHash)
+      if (head) {
```

Comment:
do we need to do this if check?

Reply:
What about an assert instead? If `head` is set in the meta table, but there's no corresponding header, we need to handle that case somehow so we don't set `this.head` to null.

Reply:
ah an assert sounds good